PeraWallet/Scenes/SwapV2/Views/SwapView.swift

```diff
@@ -161,7 +162,20 @@ struct SwapView: View {
     private func sheetContent(for sheet: SwapViewSheet) -> some View {
         switch sheet {
         case .settings:
-            SwapSettingsSheet()
+            SwapSettingsSheet(slippageSelected: $viewModel.slippageSelected) { newPercentageSelected, newSlippageSelected in
+                if let newPercentageSelected {
+                    viewModel.isLoadingPayAmount = true
+                    onAction?(.calculatePeraFee(forAmount: NSDecimalNumber(decimal: viewModel.selectedAssetIn.asset.decimalAmount).doubleValue, withPercentage: newPercentageSelected.value))
+                }
+                
+                if viewModel.slippageSelected != newSlippageSelected {
+                    
+                    viewModel.slippageSelected = newSlippageSelected?.value ?? 0 > 0 ? newSlippageSelected : nil
+                    if viewModel.shouldShowSwapButton {
+                        viewModel.updatePayingText(viewModel.payingText) { onAction?(.getQuote(for: $0)) }
+                    }
+                }
```

Comment:
Please avoid building pyramids of doom. Consider using `guard` instead of `if` if necessary.